pkg/routeagent_driver/handlers/kubeproxy_iptables/endpoint_handler.go

```diff
@@ -0,0 +1,187 @@
+package kubeproxy_iptables
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/util"
+)
+
+func (kp *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	kp.localCableDriver = endpoint.Spec.Backend
+
+	// We are on nonGateway node
+	if endpoint.Spec.Hostname != kp.hostname {
+		// If the node already has a vxLAN interface that points to an oldEndpoint
+		// (i.e., during gateway migration), delete it.
+		if kp.vxlanDevice != nil && kp.vxlanDevice.activeEndpointHostname != endpoint.Spec.Hostname {
```

Comment:
ok to merge as long as we follow up moving this to use the TransitionToGateway/NonGw logic before release